src/btree/row_modify.c

```diff
@@ -113,9 +113,18 @@ __wt_row_modify(WT_CURSOR_BTREE *cbt, const WT_ITEM *key, const WT_ITEM *value,
         } else {
             upd_size = __wt_update_list_memsize(upd);
 
+            /* If there are existing updates, append them after the new updates. */
+            if (cbt->compare == 0) {
+                for (last_upd = upd; last_upd->next != NULL; last_upd = last_upd->next)
+                    ;
+                if (cbt->ins != NULL)
+                    last_upd->next = cbt->ins->upd;
+                else if ((mod = cbt->ref->page->modify) != NULL && mod->mod_row_update != NULL)
+                    last_upd->next = mod->mod_row_update[cbt->slot];
+            }
```

Comment:
I think you can replace all of this with:
```c
                for (last_upd = upd; last_upd->next != NULL; last_upd = last_upd->next)
                     ;
                last_upd->next = *upd_entry;
```

Reply:
Also looking at this, I think there is an existing bug if there is an error: we end up calling `__wt_free_update_list` in the calling code, which will free the updates that are already attached to the tree.

In other words, we need to set `last_upd->next = NULL` on error before returning.

Reply:
Done

Reply:
You don't need to re-check `cbt->compare == 0` either.